### PR TITLE
Add SatFlowDataModule

### PR DESCRIPTION
**Pull request by jacobbieker**
# Pull Request

## Description

Adds in a datamodule for the SatFlowDataset

Fixes #

## How Has This Been Tested?

Unit test

- [ ] No
- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: nowcasting_dataloader/datamodules.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+import torch
+from nowcasting_dataset.config import load_yaml_configuration
+from nowcasting_dataset.config.model import Configuration
+from pytorch_lightning import LightningDataModule
+
+from nowcasting_dataloader.datasets import SatFlowDataset, worker_init_fn
+
+_LOG = logging.getLogger(__name__)
+_LOG.setLevel(logging.DEBUG)
```
Comment (peterdudfield) [inline nowcasting_dataloader/datamodules.py:14]: might want to remove this, let hte user decide what they want

File: nowcasting_dataloader/datamodules.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    split, transform and process the data.
+    Read the docs:
+        https://pytorch-lightning.readthedocs.io/en/latest/extensions/datamodules.html
+    """
+
+    def __init__(
+        self,
+        temp_path: str,
+        configuration: Union[Configuration, str],
+        n_train_data: int = 24900,
```
Comment (peterdudfield) [inline nowcasting_dataloader/datamodules.py:37]: we can now get this from the configruation

**Review by peterdudfield — APPROVED** [review]
small change, but looks good

**Comment by peterdudfield** [comment]
Good to wait for tests to pass - I'm sure you were going to do this anyway

File: nowcasting_dataloader/datamodules.py (hunk 1 above)
Comment (jacobbieker) [reply nowcasting_dataloader/datamodules.py:14]: Okay, sounds good!

File: nowcasting_dataloader/datamodules.py (hunk 2 above)
Comment (jacobbieker) [reply nowcasting_dataloader/datamodules.py:37]: Ah okay, cool

**Pull request merged**